R/driver_rds_remote.R

```diff
@@ -0,0 +1,199 @@
+R6_driver_rds_remote <- R6::R6Class(
+  "driver_rds_s3",
+  public = list(
+    ## TODO: things like hash_algorithm: do they belong in traits?
+    ## This needs sorting before anyone writes their own driver!
+    file_ops = NULL,
+    path = NULL,
+    compress = NULL,
+    mangle_key = NULL,
+    mangle_key_pad = NULL,
+    hash_algorithm = NULL,
+    traits = list(accept = "raw"),
+
+    initialize = function(file_ops, path, compress, mangle_key, mangle_key_pad,
+                          hash_algorithm) {
+
+      is_new <- !file_ops$object_exists(path = file.path(path, "config"))
+      file_ops$create_dir(path = path)
+      file_ops$create_dir(path = file.path(path, "data"))
+      file_ops$create_dir(path = file.path(path, "keys"))
+      file_ops$create_dir(path = file.path(path, "config"))
+
+      self$file_ops <- file_ops
+      self$path <- path
+
+      ## This is a bit of complicated dancing around to mantain
+      ## backward compatibility while allowing better defaults in
+      ## future versions.  I'm writing out a version number here that
+      ## future versions of driver_rds can use to patch, warn or
+      ## change behaviour with older versions of the storr.
+      if (!is_new && !file_ops$object_exists(path = storr:::driver_rds_config_file(path, "version"))) {
```

Comment:
You almost certainly want to get all the options into a single object and do that in one read/write operation.  I think I do that with the redis or DBI driver?

Reply:
I'll take a look and try to understand it better myself, but will hold off changing anything internal for now, given your other comments

Reply:
Hold off on this and I'll see if I can sort something out - this is going to be a headache for all the remote rds storrs

Reply:
this is fixed on the other prs now